File: binary-search/binary-search-recursive.py
def binary_search_recursive(arr, target, left, right):
    if (left <= right):
        mid = (left + right) // 2
        print('yo im here', mid)
        if arr[mid] == target:
            return mid

        elif target > arr[mid]:
            return binary_search_recursive(arr, target, mid+1, right)

        else:
            return binary_search_recursive(arr, target, left, mid-1)
    else:
        return -1



arr = [1,3,4,6,7,8,323,359,555,1000]

result = binary_search_recursive(arr, 6, 0, len(arr) - 1)

print(result)

File: binary-search/binary_search_recursive.go
package main

import (
	"fmt"
)

func BinarySerach(arr []int, target int, left int, right int) int {
	if left > right {
		return -1
	}
	mid := (left + right) / 2
	if arr[mid] == target {
		return mid
	} else if arr[mid] > target {
		return BinarySerach(arr, target, left, mid-1)
	} else {
		return BinarySerach(arr, target, mid+1, right)
	}
}

func main() {
	arr := []int{2, 4, 6, 78, 99, 344, 653, 1000, 55555}
	searchedIndex := BinarySerach(arr, 1000, 0, len(arr)-1)
	fmt.Println(searchedIndex)
}
